AudioKitSynthOne/Presets/PresetsViewController/Presets+LoadSaveManipulate.swift

```diff
@@ -365,8 +365,10 @@ extension PresetsViewController {
 
         // Pick random Preset
         var newIndex = randomNumbers.nextInt()
-        if newIndex == currentPreset.position { newIndex = randomNumbers.nextInt() }
-        currentPreset = presets[newIndex]
+        if newIndex == currentPreset.position {
+            newIndex = randomNumbers.nextInt()
```

Comment:
I am curious to know why `newIndex`  has been assigned the same value `randomNumbers.nextInt()` twice?

Reply:
I assume because we don't want the users to see "no change" in case the random index is the same one as the currently selected.

@marcussatellite might be on the 100% safe side if you replace the `if` with a `while` loop (on the off-chance the second random number is also equal)

Reply:
good callout...fixed  here